odoo/models.py

```diff
@@ -2932,6 +2932,9 @@ def read(self, fields=None, load='_classic_read'):
                         stored_fields.add(f.name)
         self._read(stored_fields)
 
+        return self._read_format(fields=fields, load=load)
+
+    def _read_format(self, fields=None, load='_classic_read'):
```

Comment:
I don't understand "read_format" naming. How is any formatting linked with the method itself ? As it seems that _read populates the cache, maybe we could imagine something like _read_from_cache ? Just proposing :) .

Reply:
The main reason of extracting the method is that is does `convert`, which calls `convert_to_read` on every field.

This is important in our case because for relational fields we want to return a pretty list of ids and names, not a recordset. Some other fields also have specific formatting.

So in a way it is "formatting" the result of a `read` for a specific purpose (sending to JS).

Calling it `_read_from_cache` would be ok for me too, considering we implicitly think of `read` as already "formatting" the result too.

Note: I changed the signature of the method since your comment because not giving `fields` is not supported here, it can't default to None.